### DISCUSSION TITLE
why autocrop ignores script-opts settings inside mpv.conf?

### DISCUSSION THREAD

**Discussion by geextahslex** [mpv-player/mpv#14500] [Q&A] [answered]
Hi, I would like to know why autocrop ignores `script-opts-append=autocrop-auto=no` that is inside mpv.conf
There are no error messages etc. Can someone help me with that?

logfile:
[output.txt](https://github.com/user-attachments/files/16132350/output.txt)

Thank you :)

**Comment by guidocella** [comment]
We would have to pass an empty callback to `read_options` to observe script-opts changes, but it wouldn't fix this case since by the time a file is loaded and profile conditions are evaluated autocrop.lua will have already enabled the crop timeout.

**Reply by geextahslex** [reply]
I don't understand. The instruction says:
-- Whether to automatically apply crop at the start of playback. If you
-- don't want to crop automatically, add
-- `script-opts-append=autocrop-auto=no` to mpv.conf.

So isn't that supposed to work?

**Reply by guidocella** [reply]
It works if you set it at startup, not in conditional profiles.

**Reply by geextahslex** [reply]
I don't know what is meant by "at startup" it is inside mpv.conf, not in a conditional profile

**Reply by guidocella** [reply]
Sorry, I was thinking about your previous questions about disabling it in a conditional profile. `script-opts-append=autocrop-auto=no` should work, I have no idea why cropdetect is still inserted for you.

**Reply by guidocella** [reply, marked as answer]
You're appending to `script-opts` and then overwriting `script-opts`' value:

```
[   0.005][v][cplayer] Setting option 'script-opts-append' = 'autocrop-auto=no' (flags = 4)
[   0.005][v][cplayer] Setting option 'script-opts' = 'ytdl_hook-try_ytdl_first=yes,ytdl_hook-exclude="%.webm$|%.ts$|%.mp3$|%.m3u8$|%.m3u$|%.mkv$|%.mp4$|%.VOB$"' (flags = 4)
```

You need to swap these 2 lines or include `autocrop-auto=no` in `script-opts=`.
